### PR TITLE
move static channelmanager functions into their own file

### PR DESCRIPTION
**Pull request by Evanfeenstra**
- new `ln::onion_payment` file with static functions from `channelmanager.rs`.
- cleaned up `test_peel_payment_onion` and moved into `ln::onion_payment`.
- made `PendingHTLCInfo.payment_hash` public (mistake from https://github.com/lightningdevkit/rust-lightning/pull/2700).
- made `UpdateAddHTLC.onion_routing_packet` public, so that `peel_payment_onion` can be used externally.
- some better comments on newly public fields.

### CODE REVIEW

**Comment by jkczyz** [comment]
I think `channelmanager_utils` is too broad. Let's use a more restrictive name lest it become another grab bag of utilities, even if only related to `ChannelManager`. Examples of other modules that we've refactored out are `inbound_payment` and `outbound_payment`. Maybe `onion_payment` in contrast to `onion_message`? Open to other ideas.

**Review by shaavan — COMMENTED** [review]
Concept ACK

The refactoring looks fantastic on my initial review, and the additional comments are really helpful! I'll take another pass at the added code to see if there are any further improvements we can make to this pull request. Superb work!

File: lightning/src/ln/channelmanager.rs [hunk 1]
```diff
@@ -118,14 +119,19 @@ pub enum PendingHTLCRouting {
...
 		/// Payment secret and total msat received.
 		payment_data: msgs::FinalOnionHopData,
 		/// See [`RecipientOnionFields::payment_metadata`] for more info.
 		payment_metadata: Option<Vec<u8>>,
+		/// CLTV expiry of the received HTLC.
 		/// Used to track when we should expire pending HTLCs that go unclaimed.
 		incoming_cltv_expiry: u32,
-		/// Optional shared secret for phantom node.
+		/// Shared secret derived using a phantom node secret key. If this field is Some, the
+		/// payment was sent to a phantom node (one hop behond the current node), but can be
```
[the remark this answers is not in the document]
Comment (Evanfeenstra) [reply lightning/src/ln/channelmanager.rs]: hehe thanks for catching that!

**Comment by Evanfeenstra** [comment]
> I think `channelmanager_utils` is too broad. Let's use a more restrictive name lest it become another grab bag of utilities, even if only related to `ChannelManager`. Examples of other modules that we've refactored out are `inbound_payment` and `outbound_payment`. Maybe `onion_payment` in contrast to `onion_message`? Open to other ideas.

Good point, I renamed the module to `onion_payment`

File: lightning/src/ln/channelmanager.rs [hunk 2]
```diff
@@ -118,14 +119,19 @@ pub enum PendingHTLCRouting {
 		short_channel_id: u64, // This should be NonZero<u64> eventually when we bump MSRV
 	},
 	/// An HTLC paid to an invoice we generated.
```
Comment (shaavan) [inline lightning/src/ln/channelmanager.rs:121]: Small nit:

In the following line, we are saying we don't know yet if the invoice was generated by us. So adding a "supposedly" will make the comments more coherent here.

```suggestion
	/// An HTLC paid to an invoice (supposedly) generated by us.
```

File: lightning/src/ln/onion_payment.rs [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+		skimmed_fee_msat: counterparty_skimmed_fee_msat,
+	})
+}
+
+/// Peel one layer off an incoming onion, returning [`PendingHTLCInfo`] (either Forward or Receive).
+/// This does all the relevant context-free checks that LDK requires for payment relay or
+/// acceptance. If the payment is to be received, and the amount matches the expected amount for
+/// a given invoice, this indicates the [`msgs::UpdateAddHTLC`], once fully committed in the
+/// channel, will generate an [`Event::PaymentClaimable`].
+///
```
Comment (shaavan) [inline lightning/src/ln/onion_payment.rs:187]: nit by linter:

```suggestion
///
```

File: lightning/src/ln/onion_payment.rs [hunk 4]
```diff
@@ -0,0 +1,499 @@
+//! Utilities for channelmanager.rs
+//!
```
Comment (shaavan) [inline lightning/src/ln/onion_payment.rs:2]: nit by linter:

```suggestion
//!
```

**Review by shaavan — COMMENTED** [review]
ACK mod nit.

This is a refactoring PR that moves static channelmanger functions to their own file.

I have used the following line command to notice the essential differences introduced by refactoring:

```bash
git diff HEAD^ --color-moved-ws=ignore-all-space --color-moved=dimmed-zebra
```

**Essential Changes in the PR:**

1. Updated comments for members of `PendingHTLCRouting` enum. The comments look clean and clear and present a better understanding of the member variables. 
2. Modified the visibility of moved code to pub(super). Since these are utilities of `channelManager` and are not meant to be used in the rest of the codebase, this visibility is the most apt one for it.
3. Removed `create_payment_onion` function. Instead, use the `create_payment_onion` function defined in `ln::onion_utils::create_payment_onion`;

The refactoring looks clean and clear, and other than the small nits, this PR is good to be merged.

**Comment by Evanfeenstra** [comment]
ok, rebased to `main` and incorporated the nits by @shaavan

**Review by shaavan — APPROVED** [review]
Code review ACK

**Review by valentinewallace — APPROVED** [review]
Fine to save these for follow-up if we want to get the initial code move in.

**Comment by valentinewallace** [comment]
Will wait til @TheBlueMatt ACK's the docs to merge.

**Comment by valentinewallace** [comment]
Going to land this since it's largely a code move and we can continue iterating on the docs if necessary in #2752.

**Pull request merged**